src/main/java/io/temporal/internal/statemachines/MutableSideEffectStateMachine.java

```diff
@@ -0,0 +1,298 @@
+                  State.REPLAYING,
+                  ExplicitEvent.SCHEDULE,
+                  State.MARKER_COMMAND_CREATED_REPLAYING,
+                  InvocationStateMachine::createFakeCommand)
+              .add(
+                  State.MARKER_COMMAND_CREATED,
+                  CommandType.COMMAND_TYPE_RECORD_MARKER,
+                  State.RESULT_NOTIFIED,
+                  InvocationStateMachine::notifyCachedResult)
+              .add(
+                  State.SKIPPED,
+                  CommandType.COMMAND_TYPE_RECORD_MARKER,
+                  State.SKIPPED_NOTIFIED,
+                  InvocationStateMachine::cancelCommandNotifyCachedResult)
+              .add(
+                  State.RESULT_NOTIFIED,
+                  EventType.EVENT_TYPE_MARKER_RECORDED,
+                  State.MARKER_COMMAND_RECORDED)
+              .add(
+                  State.MARKER_COMMAND_CREATED_REPLAYING,
+                  CommandType.COMMAND_TYPE_RECORD_MARKER,
+                  State.RESULT_NOTIFIED_REPLAYING)
+              .add(
+                  State.RESULT_NOTIFIED_REPLAYING,
+                  ExplicitEvent.NON_MATCHING_EVENT,
+                  State.SKIPPED_NOTIFIED,
+                  InvocationStateMachine::cancelCommandNotifyCachedResult)
+              .add(
+                  State.RESULT_NOTIFIED_REPLAYING,
+                  EventType.EVENT_TYPE_MARKER_RECORDED,
+                  new State[] {State.MARKER_COMMAND_RECORDED, State.SKIPPED_NOTIFIED},
+                  InvocationStateMachine::notifyFromEvent);
+
+  /** Represents a single invocation of mutableSideEffect. */
+  private class InvocationStateMachine
+      extends EntityStateMachineInitialCommand<State, ExplicitEvent, InvocationStateMachine> {
+
+    private final Functions.Proc1<Optional<Payloads>> resultCallback;
+    private final Functions.Func1<Optional<Payloads>, Optional<Payloads>> func;
+
+    InvocationStateMachine(
+        Functions.Func1<Optional<Payloads>, Optional<Payloads>> func,
+        Functions.Proc1<Optional<Payloads>> callback,
+        Functions.Proc1<StateMachine> stateMachineSink) {
+      super(
+          STATE_MACHINE_DEFINITION,
+          MutableSideEffectStateMachine.this.commandSink,
+          stateMachineSink);
+      this.func = Objects.requireNonNull(func);
+      this.resultCallback = Objects.requireNonNull(callback);
+    }
+
+    State getExecutionState() {
+      return replaying.apply() ? State.REPLAYING : State.EXECUTING;
+    }
+
+    @Override
+    public WorkflowStateMachines.HandleEventStatus handleEvent(
+        HistoryEvent event, boolean hasNextEvent) {
+      if (event.getEventType() != EventType.EVENT_TYPE_MARKER_RECORDED
+          || !event
+              .getMarkerRecordedEventAttributes()
+              .getMarkerName()
+              .equals(MUTABLE_SIDE_EFFECT_MARKER_NAME)) {
+        explicitEvent(ExplicitEvent.NON_MATCHING_EVENT);
+        return WorkflowStateMachines.HandleEventStatus.NOT_MATCHING_EVENT;
```

Comment:
so we have NOT_MATCHING_EVENT and NON_MATCHING_EVENT. Can we align these?

Reply:
Fixed.